packages/@aws-cdk/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-public.ts

```diff
@@ -0,0 +1,19 @@
+import * as path from 'path';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as ecs from '@aws-cdk/aws-ecs';
+import { App, Stack } from '@aws-cdk/core';
+
+import { QueueProcessingFargateService } from '../../lib';
+
+const app = new App();
+const stack = new Stack(app, 'aws-ecs-patterns-queue-public');
+const vpc = new ec2.Vpc(stack, 'VPC');
+
+new QueueProcessingFargateService(stack, 'PublicQueueService', {
+  vpc,
+  memoryLimitMiB: 512,
+  image: new ecs.AssetImage(path.join(__dirname, '..', 'sqs-reader')),
+  assignPublicIp: true,
```

Comment:
I wonder if we could just add this line to the existing `integ.queue-processing-fargate-service.ts`, since there's not much different between the two tests?

Reply:
There is also the fact that public subnets will be selected only if the `assignPublicIp` is `true`. They are actually two different use cases, one is for having it without any internet access and another with public internet access assigning public IPs.

I'm not sure if there is any downside on having more integration test, like if it's time consuming at some point of the CDK pipeline. If it only validate the synthesized pipeline having both use cases is helpful to further debug issues.

If at some point it require that all integ test are actually deployed would be best to consolidated all the features the public IP use case, because the VPC endpoints on the other test take some time to deploy.